src/agent/configuration_parser/CMakeLists.txt

```diff
@@ -28,18 +32,27 @@ MACRO_ENSURE_OUT_OF_SOURCE_BUILD(
 message(STATUS "CMAKE_SOURCE_DIR: ${CMAKE_SOURCE_DIR}")
 message(STATUS "CMAKE_CURRENT_SOURCE_DIR: ${CMAKE_CURRENT_SOURCE_DIR}")
 
-#########################################################################
-# Dependencies
-#########################################################################
-
-# Enable testing
-enable_testing()
-
 include_directories(include)
 include_directories(${CMAKE_SOURCE_DIR})
 
-# Add file to src code
-add_executable(ConfigurationParser_test
-    tests/configuration_parser_test.cpp
-    src/configuration_parser.cpp
-)
+find_package(GTest CONFIG REQUIRED)
```

Comment:
This package is only needed when building tests, it should be moved inside the if block below.